File: src/main/java/pl/polsl/projectmanagementsystem/model/Student.java
package pl.polsl.projectmanagementsystem.model;

import lombok.*;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.Cascade;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Student {

    @Id
    private String albumNo;

    private String userId;
    private String firstName;
    private String lastName;
    private LocalDate birthDate;

    @OneToMany(mappedBy = "student", cascade = CascadeType.ALL)
    private List<StudentGroup> studentGroupList;

    @OneToMany(mappedBy = "student", cascade = CascadeType.ALL)
    private List<StudentSemester> studentSemesterList;

    @ColumnDefault("true")
    private Boolean isActive;
}

File: src/app/model/student.ts
import { StudentGroup } from './student-group';
import { StudentSemester } from './student-semester';

export class Student {
  albumNo?: string;
  firstName: string;
  lastName: string;

  birthDate: Date;

  studentGroupIdList?: number[];
  studentSemesterIdList?: number[];
  constructor() {
    this.firstName = '';
    this.lastName = '';
    this.birthDate = new Date();
  }
}
